### PR TITLE
Fade edition bug fix

### PR DESCRIPTION
**Pull request by Zvork**
This PR fixes two small bugs when editing fade events with the *debugFade.js* script in *scripts\developer\utilities\render* concerning the "invert" and "Base gradient size" parameters. Testers should refer to the description of PR #10964 and #11021 to understand the gradient based fade effect.

# TEST PLAN
1. With the "debugFade.js" script running, checking the "Edit Fade" checkbox should apply the event selected to the combobox at the right side of the checkbox on the object in front of you.
2. With the "debugFade.js" script running, the "Edit Fade" checkbox checked and any event selected except the last "change avatar", modifying the "Invert" parameter of the edit window should modify the fade visuals by inverting the transparency
3.  With the "debugFade.js" script running, the "Edit Fade" checkbox checked and any event selected except the last two events, "user enters/leaves" and "change avatar", modifying the "Base gradient size" X, Y or Z parameters of the edit window with the "Base gradient level" set to a value greater than 0.0 should modify the fade visuals accordingly. Turning down the "Noise gradient level" to zero can help visualize this better.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2473/HighFidelity-Beta-PR11184-358373aeac29957109f0d420e930ff98f34a902c.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2473/HighFidelity-Beta-PR11184-358373aeac29957109f0d420e930ff98f34a902c.exe

File: libraries/render-utils/src/FadeEffectJobs.cpp [hunk 1]
```diff
@@ -594,7 +594,7 @@ void FadeJob::run(const render::RenderContextPointer& renderContext, FadeJob::Ou
         if (update(*jobConfig, scene, transaction, state, deltaTime)) {
             hasTransaction = true;
         }
-        if (isFirstItem) {
+        if (isFirstItem && jobConfig->manualFade && state.threshold!=jobConfig->threshold) {
```
Comment (samcake) [inline libraries/render-utils/src/FadeEffectJobs.cpp]: can you put parenthesis around the != test
Also there should be spaces around the != 
Thanks

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2484/HighFidelity-Beta-PR11184-c88082301e9a8fe8b9df298f807216650748f55c.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2484/HighFidelity-Beta-PR11184-c88082301e9a8fe8b9df298f807216650748f55c.exe

**Comment by Nex-Pro** [comment]
Looking at this

**Comment by Nex-Pro** [comment]
With the "debugFade.js" script running, checking the "Edit Fade" checkbox should apply the event selected to the combobox at the right side of the checkbox on the object in front of you. **SUCCESS**
With the "debugFade.js" script running, the "Edit Fade" checkbox checked and any event selected except the last "change avatar", modifying the "Invert" parameter of the edit window should modify the fade visuals by inverting the transparency **SUCCSESS**
With the "debugFade.js" script running, the "Edit Fade" checkbox checked and any event selected except the last two events, "user enters/leaves" and "change avatar", modifying the "Base gradient size" X, Y or Z parameters of the edit window with the "Base gradient level" set to a value greater than 0.0 should modify the fade visuals accordingly. **SUCCESS**

Turning down the "Noise gradient level" to zero can help visualize this better.

There is an issue with textures or possibly lighting that I am seeing. After the effect appears to have completed some entities appear to be extremely dark or maybe black.

See image.  Compare the two plants.  They should look more or less the same. The one on the left is too dark. It is the one that has had the fade effect. Second image the one on right has had effect. Third image neither has had effect.

hifi-snap-by--on-2017-08-15_18-56-38


hifi-snap-by--on-2017-08-15_19-00-09

hifi-snap-by--on-2017-08-15_19-00-23

**Comment by Zvork** [comment]
@Nex-Pro Nice catch! I'll take a look.

**Comment by Zvork** [comment]
Concerning the bug on the plant model detected by @Nex-Pro , after investigation it is linked to the way the object is modeled. Turning on the fade effect on a model switches its rendering to double sided. Unfortunately this plant has already double the geometry to account for the thin geometry of the leaves, one set for each normal direction. When double sided rendering is activated (which is the case when fade is running, again), in effect each side of a leaf is rendered twice and there is a high probability that it is with the wrong normal, thus the broken lighting.
After discussion with @samcake , we will leave it as is in this PR and we will need to address this more general problem (same thing happens with the tree leaves) in another PR probably during the baking phase when importing models.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2565/HighFidelity-Beta-PR11184-9691c957e49c4b394558b982543122d0d3a40279.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2565/HighFidelity-Beta-PR11184-9691c957e49c4b394558b982543122d0d3a40279.exe

**Pull request merged**